ADAL/src/cache/ADTokenCache.m

```diff
@@ -69,6 +69,45 @@ - (void)dealloc
     SAFE_ARC_SUPER_DEALLOC();
 }
 
+- (id)cacheObjectForKey:(id<NSCopying>)key
+{
+    id object = nil;
+    @synchronized(self)
```

Comment:
What's the alternative? I could add the `@synchronize` calls inline, but it doesn't remove the need for the mutex. We'd still need to add the `@synchronize`s, unless you have a different strategy?